### DISCUSSION TITLE
Make the Repository collaboration-ready

### DISCUSSION THREAD

**Discussion by nicolaerario** [Il-Libro-Open-Source/book#23] [Idee]
Butto giù una bozza di `checklist` per rendere il repository pronto alla collaborazione.
_Collaboriamo per rendere la lista definitiva_ 😎

### 1. Repository Settings

- [x]  set appropriate repository visibility (public or private)
- [ ]  configure branch protection rules

### 2. Repository Contents

- [x]  create an informative README.md
- [ ]  add a LICENSE.md file
- [ ]  include CONTRIBUTING.md
- [ ]  set up CODEOWNERS
- [x]  establish a CODE_OF_CONDUCT.md

### 3. Automation/Checks

- [x]  implement code linting / formatting
- [ ]  set up tests / Continuous Integration (CI)
- [ ]  configure status checks
- [ ]  set up Continuous Deployment (CD)

### 4. Security

- [ ]  assign appropriate roles / permissions
- [ ]  manage secrets securely
- [ ]  set up scanners for dependencies, secrets, general vulnerabilities
- [ ]  setup Dependabot version updates
- [ ]  provide a SECURITY.md

### 5. Advanced Options

- [ ]  add Issue Templates
- [ ]  define Environments

### 6. Next Steps / Externalities

- [ ]  commit to being responsive
- [ ]  use GitHub Projects and Issues for project management
- [ ]  promote the project for visibility
- [ ]  engage with the community

**Comment by nicolaerario** [comment]
Ho voluto rendela più generale possibile di proposito, naturalmente alcuni punti (come il 3 ad esempio) dipendono da decisioni future.

**Reply by Cadienvan** [reply]
Ciao Nicola, sì come dici tu probabilmente alcuni di questi punti non ci serviranno, dubito possa servirci una CI per un libro ahahah però ottima idea, teniamola sotto mano!

**Reply by nicolaerario** [reply]
Perfetto grazie. Intanto io ho messo, c’è sempre tempo per togliere. E poi, vuoi mettere la soddisfazione di automatizzare le build dei capitoli, indice ecc? 😎

**Reply by Cadienvan** [reply]
Sky is the limit!

**Reply by francescobianco** [reply]
@nicolaerario @Cadienvan al contrario sono convito che invece una CI sia proprio necessaria per un libro, soprattutto se nasce in questo contesto. C'è bisogno di dare l'esempio e mostrare come anche un libro sia un artefatto che può essere processato da processi automatici per la qualità. Ecco alcuni esempi da considerare per la CI/CD.

- Controllo automatico lessicale e grammaticale
- Generazione automatica di indici e bibliografie
- Generazione e distribuzione degli artefatti finali, es. versione in PDF, ePUB, etc...

**Reply by Cadienvan** [reply]
Mi piace questa verve! Ahah va bebe, se po' fa'!

**Reply by nicolaerario** [reply]
@francescobianco sfondi una porta aperta con me

**Comment by nicolaerario** [comment]
Evoluto in issue #56
